android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1263,6 +1217,39 @@ public Tab openNewOrSelectExistingTab(String url) {
         }
     }
 
+    private void setupWalletModel() {
+        if (walletModel == null) {
+            walletModel = new WalletModel(mKeyringService, mBlockchainRegistry, mJsonRpcService,
+                    mTxService, mEthTxManagerProxy, mSolanaTxManagerProxy, mAssetRatioService,
+                    mBraveWalletService);
+        } else {
+            walletModel.setKeyringService(mKeyringService);
```

Comment:
nit: better to create method for this in `WalletModel` which takes all these services and set them.

Reply:
The initial idea was to only set the required ( `null` ) services (but not ideal in this case) and to not violate "too many parameter" rule of thumb (with constructor we have no choice) but I guess there's no better approach than having a method with same params as constructor?